Chartboost/src/main/java/com/applovin/mediation/adapters/ChartboostMediationAdapter.java

```diff
@@ -509,100 +516,179 @@ public void didDismissRewardedVideo(String location)
 
                     // Passing extra info such as creative id supported in 9.15.0+
-                    if ( AppLovinSdk.VERSION_CODE >= 9150000 && !TextUtils.isEmpty( event.adID ) )
+                    if ( AppLovinSdk.VERSION_CODE >= 9150000 && !TextUtils.isEmpty( showEvent.getAdID() ) )
                     {
                         Bundle extraInfo = new Bundle( 1 );
-                        extraInfo.putString( "creative_id", event.adID );
+                        extraInfo.putString( "creative_id", showEvent.getAdID() );
 
-                        onAdDisplayed( location, extraInfo );
+                        ChartboostMediationAdapterRouter.this.onAdDisplayed( location, extraInfo );
                     }
                     else
                     {
-                        onAdDisplayed( location );
+                        ChartboostMediationAdapterRouter.this.onAdDisplayed( location );
                     }
                 }
             }
 
             @Override
-            public void onAdClicked(final ChartboostClickEvent event, final ChartboostClickError error)
-            {
-                String location = adView.getLocation();
-                if ( error != null )
+            public void onAdRequestedToShow(@NonNull ShowEvent showEvent) {
+                log( "AdView onAdRequestedToShow: " + showEvent.getAd().getLocation() );
+            }
+
+            @Override
+            public void onAdLoaded(@NonNull CacheEvent cacheEvent, @Nullable CacheError cacheError) {
+                String location = cacheEvent.getAd().getLocation();
+                if ( cacheError != null )
                 {
-                    log( "Failed to record click on \"" + location + "\" because of error: " + error.code );
+                    MaxAdapterError adapterError = toMaxError( cacheError );
+
+                    log( "AdView failed \"" + location + "\" to load with error: " + cacheError.getCode() );
+                    onAdLoadFailed( location, adapterError );
+                }
+                else
+                {
+                    log( "AdView loaded: " + location );
+
+                    // Passing extra info such as creative id supported in 9.15.0+
+                    if ( AppLovinSdk.VERSION_CODE >= 9150000 && !TextUtils.isEmpty( cacheEvent.getAdID() ) )
+                    {
+                        Bundle extraInfo = new Bundle( 1 );
+                        extraInfo.putString( "creative_id", cacheEvent.getAdID() );
+
+                        ChartboostMediationAdapterRouter.this.onAdLoaded( location, extraInfo );
+                    }
+                    else
+                    {
+                        ChartboostMediationAdapterRouter.this.onAdLoaded( location );
+                    }
+
+                    showAdViewDelayed();
+                }
+            }
+
+            @Override
+            public void onAdClicked(@NonNull ClickEvent clickEvent, @Nullable ClickError clickError) {
```

Comment:
To the reviewer: During internal tests we found odd behaviour. In the adapter we reach banner onAdClick callback and we call ChartboostMediationAdapterRouter.this.onAdClicked( location ); but then this callback is not forwarded back to the caller which is applovin MaxAdView and listener implemented is: override fun onAdClicked(ad: MaxAd?)

Is this a bug or something wrong with our implementation? Interstitial and Rewarded forwards the callbacks as expected upon click.

Reply:
It's your implementation - we have a router architecture specifically for your SDK due to the singleton callback but now you've moved over to a new set of more deterministic APIs. We're working/testing this on our end accordingly.